### PR TITLE
Adding full test suite on a weekly basis

### LINKED ISSUE #1752: Run full test suite between releases
Currently we only run the 'normal' tests on all security systems nightly and automated. This means that if there is an issue which effects one of the extended tests (node versions, or keyring config), we might not find out about it until RC generation, which will lead to delayed releases ( as an example we currently have broken keyring support in staging).

I think it would be at least worth running all tests at the weekend, on the opposite night that performance tests are run, so we can hopefully find these regressions in a week, rather than month.

If we can implement some of the changes in #1751 then we might be able to increase this further.

### PR DESCRIPTION
**Pull request by FlappiTomic**
Please check if your PR fulfills the following requirements. This is simply a reminder of what we are going to look for before merging your PR. If you don't know all of this information when you create this PR, don't worry. You can edit this template as you're working on it.

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Necessary documentation (if appropriate) have been added / updated
- [x] DCO signoffs have been added to all commits, including this PR

#### PR type
What type of changes does your PR introduce to Zowe? _Put an `x` in the box that applies to this PR. If you're unsure about any of them, don't hesitate to ask._ 
- [ ] Bugfix 
- [x] Feature 
- [ ] Other... Please describe:

#### Relevant issues

Fixes #1752

#### Changes proposed in this PR

- 1. Adding multiple node versions bundle and multiple keystore modes bundle to run on Saturdays
- 2. Rename "multiple keystore modes" to "multiple keyring modes" and its references

#### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

#### Does this PR do something the person installing Zowe should know about?

---
* Affected function: _general area of interest_ *

---
* Description: _1 line description_ *

---
* Part: _name of customizable file involved_  *

---
_multi-line description_

#### Is there a related doc issue or Pull Request? 

Doc issue/PR number:

### CODE REVIEW

File: Jenkinsfile.nightly [hunk 1]
```diff
@@ -59,6 +61,14 @@ node('zowe-jenkins-agent-dind-wdc') {
     pipeline.addBuildOption(pipelineTriggers([cron("TZ=America/New_York\nH 23 * * *")]))
   }
 
+  // get the day of the week
+  def currentDayofWeek = sh(script: 'date +"%a"', returnStdout: true).trim()
+  def fullTestRun
+  // only run full test suite on Saturdays
+  if (currentDayofWeek.equals('Sat')) {
+    fullTestRun = 'true'
```
Comment (jackjia-ibm) [inline Jenkinsfile.nightly:69]: Can use `true` directly as boolean.

```
boolean fullTestRun = false
fullTestRun = true
```

File: tests/installation/Jenkinsfile [hunk 2]
```diff
@@ -89,8 +89,8 @@ node('zowe-jenkins-agent-dind-wdc') {
       jenkins_primary_lock: 'marist',
       jenkins_extra_locks: ['marist-2', 'marist-3'],
     ],
-    'bundle: multiple keystore modes': [
-      test_files: 'dist/__tests__/extended/keystore-modes/',
+    'bundle: multiple keyring modes': [
```
Comment (jackjia-ibm) [inline tests/installation/Jenkinsfile:92]: Maybe call it `bundle: keyring on multiple security systems`?

**Review by jackjia-ibm — APPROVED** [review]
Looks great, only 2 minor points which I think are optional. Thanks a lot Tom!

File: Jenkinsfile.nightly (hunk 1 above)
Comment (FlappiTomic) [reply Jenkinsfile.nightly:69]: yeh make more sense. Changed

**Comment by jackjia-ibm** [comment]
@FlappiTomic this change is hard to verify except we merge and wait for Saturday. To mock it, we can change the date to today (Friday), comment out or skip the Promote and Message steps and kick of a nightly build run. what do you think?

**Comment by FlappiTomic** [comment]
> @FlappiTomic this change is hard to verify except we merge and wait for Saturday. To mock it, we can change the date to today (Friday), comment out or skip the Promote and Message steps and kick of a nightly build run. what do you think?

Yeah sounds good, I will do Fri temporarily

**Pull request merged**